src/main/java/duke/ui/text/TextUi.java

```diff
@@ -0,0 +1,256 @@
+        String input = this.scanner.nextLine();
+        System.out.println(TextUi.HORIZONTAL_LINE);
+        return input;
+    }
+
+    /**
+     * Invoked at the end of the programme, to leave an exit message to user.
+     */
+    @Override
+    public void exit() {
+        System.out.println("Bye. Hope to see you again soon!");
+        System.out.println(TextUi.HORIZONTAL_LINE);
+        this.scanner.close();
+    }
+
+    /**
+     * Notifies user of a user-input error.
+     */
+    @Override
+    public void notifyError(String message) {
+        System.out.println(this.errorPrepend + message + this.errorAppend);
+    }
+
+    /**
+     * Notifies user that a task has been marked done
+     * @param task the task to notify
+     */
+    @Override
+    public void notifyMarkDone(Task task) {
+        System.out.println("Nice! I've marked this task as done:");
+        System.out.println(task);
+    }
+
+    /**
+     * Notifies user that a task has been marked not done
+     * @param task the task to notify
+     */
+    @Override
+    public void notifyMarkNotDone(Task task) {
+        System.out.println("OK, I've marked this task as not done yet:");
+        System.out.println(task);
+    }
+
+    /**
+     * Notifies user that a task has been removed
+     * @param task the task removed
+     */
+    @Override
+    public void notifyRemoved(Task task) {
+        System.out.println("Noted, I've removed this task:");
+        System.out.println(task);
+    }
+
+    /**
+     * Notifies that a list of task is going to be displayed.
+     * Does not display the tasks itself.
+     * @param type type of task (todo/deadline/event/default)
+     * @param isExcludingDone whether to exclude tasks already done
+     * @param date the date before which to display deadlines before or events happening on,
+     *             null if not to filter by date
+     * @param taskList the task list to display
+     */
+    @Override
+    public void notifyList(Type type, boolean isExcludingDone, LocalDate date, TaskList taskList) {
+        String typeString;
+        switch (type) {
+        case TODO:
+            typeString = "to-do tasks";
+            break;
+        case DEADLINE:
+            typeString = "deadlines";
+            break;
+        case EVENT:
+            typeString = "events";
+            break;
+        default:
+            typeString = "tasks";
+            break;
+        }
+        System.out.println(
+                "Alright, here is your list of "
+                + typeString
+                + (isExcludingDone ? " not done" : "")
+                + (date != null
+                        ? (type == Type.DEADLINE
+                            ? " before "
+                            : type == Type.EVENT
+                            ? " happening on "
+                            : " for "
+                        ) + DateTimeManager.dateToDisplay(date)
+                        : ""
+                ) + ":"
```

Comment:
Consider breaking up this chain of ternary operators, by resolving the logic outside of the print statement first.